packages/core/src/tenants/SystemContext.ts

```diff
@@ -11,25 +20,44 @@ const { table, fields } = convertToIdentifiers(Systems);
 export default class SystemContext {
   static shared = new SystemContext();
   public storageProviderConfig: StorageProviderData | undefined;
+  public hostnameProviderConfig: HostnameProviderData | undefined;
+
+  async loadProviderConfigs(pool: CommonQueryMethods) {
+    this.storageProviderConfig = await this.loadConfig<StorageProviderData>(
+      pool,
+      StorageProviderKey.StorageProvider,
+      storageProviderDataGuard
+    );
+    this.hostnameProviderConfig = await this.loadConfig<HostnameProviderData>(
+      pool,
+      CloudflareKey.HostnameProvider,
+      hostnameProviderDataGuard
+    );
+  }
 
-  async loadStorageProviderConfig(pool: CommonQueryMethods) {
+  private async loadConfig<T>(
+    pool: CommonQueryMethods,
+    key: SystemKey,
+    guard: ZodType
+  ): Promise<T | undefined> {
     const record = await pool.maybeOne<Record<string, unknown>>(sql`
       select ${fields.value} from ${table}
-      where ${fields.key} = ${StorageProviderKey.StorageProvider}
+      where ${fields.key} = ${key}
     `);
 
     if (!record) {
       return;
     }
 
-    const result = storageProviderDataGuard.safeParse(record.value);
+    const result = guard.safeParse(record.value);
 
     if (!result.success) {
-      consoleLog.error('Failed to parse storage provider config:', result.error);
+      consoleLog.error('Failed to parse cloudflare hostname provider config:', result.error);
```

Comment:
Also here, shouldn't the error message be more generic? E.g. `Failed to parse provider config.`

Reply:
same as the answer above

Reply:
But the function name `loadConfig<T>()` seems to be generic enough. Why is it specific to load cloudflare configs?

Reply:
updated with dynamic key name